packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -244,7 +244,7 @@ class NumberInput extends Component {
         ? (min !== undefined && value > min) || min === undefined
         : (max !== undefined && value < max) || max === undefined;
 
-    if (!disabled && conditional) {
+    if (!disabled && conditional && typeof step !== 'string') {
```

Comment:
With `step` being either a number of `'any'` would it make sense to have the check be `step !== 'any'`?